frontend/app/components/import_report/import_report.component.ts

```diff
@@ -0,0 +1,241 @@
+      this._dataService.getTaxaRepartition(idSource, this.rank).subscribe((data) => {
+        this.taxaDistribution = data;
+        this.updateChart();
+      });
+    }
+  }
+
+  loadDatasetName() {
+    if (this.importData) {
+      this._dataService.getDatasetFromId(this.importData.id_dataset).subscribe((data) => {
+        this.datasetName = data.dataset_name;
+      });
+    }
+  }
+
+  loadErrors() {
+    if (this.importData) {
+      this._dataService.getImportErrors(this.importData.id_import).subscribe((errors) => {
+        this.importErrors = errors;
+        // Get the total number of errors:
+        // 1. get all rows in errors
+        // 2. flaten to have 1 array of all rows in error
+        // 3. remove duplicates (with Set)
+        // 4. return the size of the Set (length)
+        this.nbTotalErrors = new Set(
+          errors.map((item) => item.rows).reduce((acc, val) => acc.concat(val), [])
+        ).size;
+      });
+    }
+  }
+
+  matchNomenclature() {
+    if (
+      this.importData &&
+      this.importData?.contentmapping &&
+      Object.keys(this.importData?.contentmapping).length > 0
+    ) {
+      const mappingValues = this.importData.contentmapping;
+      const match: Array<MatchedNomenclature> = [];
+      this._dataService.getNomenclatures().subscribe((nomencs) => {
+        Object.keys(nomencs).forEach((nomenc) => {
+          const type_mnemo = nomencs[nomenc].nomenclature_type.mnemonique;
+          Object.keys(mappingValues[type_mnemo]).forEach((val) => {
+            const sourceNomenclature = nomencs[nomenc].nomenclatures.find(
+              (n) => n.cd_nomenclature === mappingValues[type_mnemo][val]
+            );
+            const targetNomenclature = nomencs[nomenc].nomenclatures.find(
+              (n) => n.label_default === val
+            );
+            match.push({
+              source: sourceNomenclature,
+              target: targetNomenclature,
+            });
+          });
+        });
+      });
+      this.matchedNomenclature = match;
+    }
+  }
+
+  updateChart() {
+    const labels: string[] = this.taxaDistribution.map((e) => e.group);
+    const data: number[] = this.taxaDistribution.map((e) => e.count);
+
+    this.doughnutChartLabels.length = 0;
+    // Must push here otherwise the chart will not update
+    this.doughnutChartLabels.push(...labels);
+
+    this.doughnutChartData.length = 0;
+    this.doughnutChartData = data;
+
+    // Fill colors with random colors
+    const colors: string[] = new Array(this.doughnutChartData.length);
+    for (let i = 0; i < colors.length; i++) {
+      colors[i] = '#' + (((1 << 24) * Math.random()) | 0).toString(16);
+    }
+    this.doughnutChartColors[0].backgroundColor.push(...colors);
+  }
+
+  getChartPNG(): HTMLImageElement {
+    const chart: HTMLCanvasElement = <HTMLCanvasElement>document.getElementById('chart');
+    const img: HTMLImageElement = document.createElement('img');
+    img.src = chart.toDataURL();
```

Comment:
En cliquant sur « Export PDF », erreur du front ici : ``TypeError: chart is null``

Reply:
Bien vu, il y avait un bug quand le graphique n'était pas dispo.
Merci beaucoup !